includes/Core/Permissions/Permissions.php

```diff
@@ -399,6 +414,48 @@ private function check_dashboard_sharing_capability( $cap, $user_id, $args ) {
 		}
 	}
 
+	/**
+	 * Checks if the VIEW_SPLASH capability is allowed for the user.
+	 *
+	 * Allows access to the VIEW_SPLASH capability if the user can authenticate or has
+	 * VIEW_SHARED_DASHBOARD capability.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param int $user_id User ID of the user the capability is checked for.
+	 * @return array Array with a 'do_not_allow' element if checks fail, empty array if checks pass.
+	 */
+	private function check_view_splash_capability( $user_id ) {
+		if ( Feature_Flags::enabled( 'dashboardSharing' ) ) {
+			if ( in_array( 'do_not_allow', $this->check_view_shared_dashboard_capability( $user_id ), true ) ) {
+				return array( self::AUTHENTICATE );
+			}
+			return array();
+		}
+		return array( self::AUTHENTICATE );
```

Comment:
Nit: This could be simplified a little by making the guard specific to the feature not enabled case which would swap the placement of the returns making for overall less indentation.